data_structures/heap/min_heap.py

```diff
@@ -0,0 +1,154 @@
+# Min head data structure
+# with decrease key functionality - in O(log(n)) time
+
+
+class Node:
+    def __init__(self, val, name):
+        self.val = val
+        self.name = name
+
+    def __str__(self):
+        return self.name
+
+    def __lt__(self, other):
+        return self.val < other.val
+
+
+class MinHeap:
+    """
+    >>> r = Node(-1, "R")
+    >>> b = Node(6, "B")
+    >>> a = Node(3, "A")
+    >>> x = Node(1, "X")
+    >>> e = Node(4, "E")
+    >>> myMinHeap = MinHeap([r, b, a, x, e])
+    >>> myMinHeap.decrease_key(b, -17)
+    """
```

Comment:
```
>>> print(b)
Node('b', -17)
```

Reply:
I have added this and btw it it is 
`>>> print(b)
Node(B, -17)` 
without quotes around B

Reply:
Did this test fail?